[Term Entry] Java ArrayList .contains() (#4379)
Codecademy/docs: content/java/concepts/array-list/terms/contains/contains.md

---
Title: '.contains()'

> Description: 'To check if an element is present in the list or not.'

Reviewer: ```suggestion
Description: 'Checks if an element is present in the ArrayList or not.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Data Types'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The `.contains()` method is used for checking if the specified element is present in `ArrayList` or not. It will return true if the element is present; otherwise, it will return false.

Reviewer: ```suggestion
The **`.contains()`** method is used for checking if the specified element is present in `ArrayList` or not. It will return true if the element is present; otherwise, it will return false.
```
You should bold the first occurrence of the method name.

## Syntax

To check if an element is present in `ArrayList`, it is passed as an argument in the `.contains()` method of `ArrayList`. The return value is of the boolean type. True if the element is present; otherwise, false if it is not present. More formally, it returns true if and only if this `ArrayList` contains at least one element `e` such that `(o==null? e==null: o.equals(e))`. Here `o` is the object passed to the list, and `e` is an element of the list.

```pseudo

> arrayListInstance.contains(o);

Reviewer: ```suggestion
arrayListInstance.contains(obj);
```

Author: I wrote it `o` because I mention it in the paragraph.
`(o==null? e==null: o.equals(e))` 
I think it will be better to change `o` to `obj` in the paragraph as well.

```

> 

Reviewer: Can you describe the parameter `obj` in one line below the syntax block? Do it in the following format - 
```
- `parameter`: Its description.
```

Author: `obj`: Element whose presence in the ArrayList is to be tested

## Example

> In the example below, an empty `ArrayList` instance `fruitList` is created and can hold `String`-type elements. Next, a few fruit names are added with the [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) method. Lastly, the presence of two fruit names is checked in the `ArrayList` with the `.contains()` method:

Reviewer: ```suggestion
In the example below, an empty `ArrayList` instance `fruitList` is created and can hold `String` type elements. Next, a few fruit names are added with the [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) method. Lastly, the presence of two fruit names is checked in the `ArrayList` with the `.contains()` method:
```

```java
import java.util.ArrayList;

public class Fruits {
  public static void main(String[] args) {

    ArrayList<String> fruitList = new ArrayList<String>();

    fruitList.add("Mangos");
    fruitList.add("Bananas");
    fruitList.add("Watermelons");
    fruitList.add("Grapes");

    // Check if Oranges and Bananas are present in the fruit list or not.
    boolean areOrangesPresent = fruitList.contains("Oranges");
    boolean areBananasPresent = fruitList.contains("Bananas");

    System.out.println("Fruit list contains oranges: "+ areOrangesPresent);
    System.out.println("Fruit list contains bananas: "+ areBananasPresent);
  }
}
```

> The output should look like this:

Reviewer: ```suggestion
The output of the above will look like this:
```

---
Title: '.contains()'
Description: 'Checks if an element is present in the ArrayList or not.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Data Types'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The **`.contains()`** method is declared in the `List` interface and implemented in the `ArrayList` class. It is used to check if the element is present in the specified `ArrayList` or not. The function returns a boolean value of `true` if the element is present and `false` if not.

## Syntax

> The `.contain()` method can be called on an `ArrayList` instance, and it requires one parameter, the object whose presence we need to check in `ArrayList`. More formally, it returns `true` if and only if the specified `ArrayList` contains at least one element `e` such that `(obj==null? e==null: obj.equals(e))`. Here `obj` is the object passed to the list, and `e` is an element of the list.

Reviewer: ```suggestion
The `.contains()` method can be called on an `ArrayList` instance, and it requires one parameter, the object whose presence we need to check in `ArrayList`. More formally, it returns `true` if and only if the specified `ArrayList` contains at least one element `e` such that `(obj==null? e==null: obj.equals(e))`. Here `obj` is the object passed to the list, and `e` is an element of the list.
```
